crates/router/src/connector/adyen/transformers.rs

```diff
@@ -121,18 +132,26 @@ pub struct Address {
     country: api_enums::CountryAlpha2,
     house_number_or_name: Secret<String>,
     postal_code: Secret<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
     state_or_province: Option<Secret<String>>,
-    street: Secret<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    street: Option<Secret<String>>,
```

Comment:
Why is this made option?

Reply:
I made this optional since without making it optional, the entire address might get None if line2 is not present (according to the current code implementation)

![Screenshot 2024-12-11 at 15 52 20](https://github.com/user-attachments/assets/2c751fb4-3012-40bf-a64d-140b36e182b3)

<img width="593" alt="Screenshot 2024-12-11 at 15 51 59" src="https://github.com/user-attachments/assets/60161d0a-f388-4180-bda9-01be1d7c9c59">

Also for Adyen street is optional. (I have tested this for Paypal, Cards)